### PR TITLE
Micro-optimizations with ~16% speedup Canary training on 1GPU

### PR DESCRIPTION
**Pull request by pzelasko**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/models/aed_multitask_models.py [hunk 1]
```diff
@@ -674,6 +675,11 @@ def training_step(self, batch: PromptedAudioToTextMiniBatch, batch_nb):
 
         input_ids, labels = batch.get_decoder_inputs_outputs()
 
+        num_frames = batch.audio_lens.sum().float()
+        num_tokens = batch.prompted_transcript_lens.sum().float()
+        tot_frames = torch.as_tensor(batch.audio.numel(), device=num_frames.device, dtype=torch.float)
+        tot_tokens = torch.as_tensor(batch.prompted_transcript.numel(), device=num_frames.device, dtype=torch.float)
```
Comment (pzelasko) [inline nemo/collections/asr/models/aed_multitask_models.py:681]: puts logged metrics into float tensors so PTL doesn't have to do costly conversion; removes cuda sync on division with python floats in logged dict

File: nemo/collections/asr/models/aed_multitask_models.py [hunk 2]
```diff
@@ -683,14 +689,10 @@ def training_step(self, batch: PromptedAudioToTextMiniBatch, batch_nb):
...
-        num_frames = batch.audio_lens.sum()
-        num_tokens = batch.prompted_transcript_lens.sum()
-        tot_frames = batch.audio.numel()
-        tot_tokens = batch.prompted_transcript.numel()
         tensorboard_logs = {
             'train_loss': audio_loss,
-            'learning_rate': self._optimizer.param_groups[0]['lr'],
-            'batch_size': batch.audio.shape[0],
+            'learning_rate': torch.as_tensor(self._optimizer.param_groups[0]['lr']),
+            'batch_size': torch.as_tensor(batch.audio.shape[0]),
```
Comment (pzelasko) [inline nemo/collections/asr/models/aed_multitask_models.py:695]: avoids costly ptl conversion to tensor later

File: nemo/utils/exp_manager.py [hunk 3]
```diff
@@ -264,7 +264,7 @@ def _on_batch_end(self, name, pl_module):
         # Set the `batch_size=1` as WAR for `dataloader_iter`, which is not used for any metric
         pl_module.log(
             name + ' in s',
-            self.timer[name],
+            torch.as_tensor(self.timer[name]),
```
Comment (pzelasko) [inline nemo/utils/exp_manager.py:267]: avoids costly PTL conversion to tensor
